src/main/java/com/ays/assignment/service/impl/AssignmentSaveServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.ays.assignment.service.impl;
+
+import com.ays.assignment.model.Assignment;
+import com.ays.assignment.model.dto.request.AssignmentSaveRequest;
+import com.ays.assignment.model.entity.AssignmentEntity;
+import com.ays.assignment.model.mapper.AssignmentSaveRequestToAssignmentMapper;
+import com.ays.assignment.repository.AssignmentRepository;
+import com.ays.assignment.service.AssignmentSaveService;
+import com.ays.auth.model.AysIdentity;
+import com.ays.common.model.AysPhoneNumber;
+import com.ays.user.util.exception.AysUserAlreadyExistsByPhoneNumberException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * UserSaveServiceImpl is an implementation of the {@link AssignmentSaveService} interface.
+ * It provides methods for saving assignment data and performing related operations by admin.
+ */
+@Service
+@RequiredArgsConstructor
+class AssignmentSaveServiceImpl implements AssignmentSaveService {
+
+    private final AssignmentRepository assignmentRepository;
+
+    private final AysIdentity identity;
+
+    private static final AssignmentSaveRequestToAssignmentMapper assignmentSaveRequestToAssignmentMapper = AssignmentSaveRequestToAssignmentMapper.initialize();
+
+    /**
+     * Saves a new assignment based on the provided save request.
+     *
+     * @param saveRequest the request object containing assignment data to be saved
+     * @return the saved assignment
+     * @throws AysUserAlreadyExistsByPhoneNumberException if a assignment with the same phone number already exists
+     */
+    @Override
+    public Assignment saveAssignment(AssignmentSaveRequest saveRequest) {
+
+        final List<AssignmentEntity> assignmentsFromDatabase = assignmentRepository.findAll();
+
+        this.checkPhoneNumberExist(saveRequest.getPhoneNumber(), assignmentsFromDatabase);
+
+        final AssignmentEntity assignmentEntity = assignmentSaveRequestToAssignmentMapper.mapForSaving(saveRequest, identity.getInstitutionId());
+
+        AssignmentEntity savedAssignment = assignmentRepository.save(assignmentEntity);
+
+        return Assignment.builder()
+                .id(savedAssignment.getId())
+                .description(saveRequest.getDescription())
+                .firstName(saveRequest.getFirstName())
+                .lastName(saveRequest.getLastName())
+                .latitude(saveRequest.getLatitude())
+                .longitude(saveRequest.getLongitude())
+                .phoneNumber(saveRequest.getPhoneNumber())
+                .build();
+
+    }
```

Comment:
Bu metodun geri dönüş tipine ihtiyacımı olmayacağı için herhangi bir mapleme de yapmaya ihtiyacımız bulunmuyor, doğrudan `void` yapabiliriz

Reply:
Return type `void `olarak düzelttim.